### PR TITLE
Add & Remove task

### PR DESCRIPTION
**Pull request by nottherealalanturing**
## Describe your changes
Features added include

- Ability to add a task to Task List
- Ability to remove task to Task List
- Tasks description can be edited.

### What kind of change does this PR introduce?

New Feature - Add, Remove and Edit tasks.

### In which branch do you want the changes to merge?

Main.

### Documentation updated ?

Yes

### Checklist

- [ ] No ES Lint Error
- [ ] No Style Lint Error
- [ ] Unit test case update

### Screenshots
Screen Shot 2022-04-28 at 18 20 06

### CODE REVIEW

File: src/Tasks.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    this.tasks = JSON.parse(localStorage.getItem('tasks')) || [];
+  }
+
+  addTask = (description, completed) => {
+    this.tasks.push({
+      description,
+      completed,
+      index: parseInt(Tasks.index, 10),
+    });
+  };
```
Comment (DeliceLydia) [inline src/Tasks.js]: - [ ] When I add a task the property `index` should be set to the value of the new array (i.e. if a 5th task is added to the list, the `index` of that task should equal to 5) but currently when I check your application all indexes are 0 please address that
idx

File: src/Tasks.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+export default class Tasks {
+  static index = 0;
```
Comment (DeliceLydia) [inline src/Tasks.js]: - [ ] The issue of the index is coming from this static index it should actually be updated anytime you add a new task

File: src/Tasks.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  };
+
+  deleteTask = (index) => {
+    this.tasks.forEach((val, i) => {
+      if (this.tasks[i].index === index) {
+        this.tasks.splice(i, 1);
+      }
+      Tasks.index -= 1;
+    });
+  };
```
Comment (DeliceLydia) [inline src/Tasks.js]: - [x] This function should delete a selected task and it should update all remaining items' `indexes`, so they represent the current list order but this is what happens when I delete a task 
here is a list of tasks I have
delete
see what happens when I try to delete `shopping`
sh
it actually doesn't delete a selected one it deletes others and then return the selected task twice please address this issue

File: src/Tasks.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+  editTask = (index, description) => {
+    this.tasks.forEach((val, i) => {
+      if (this.tasks[i].index === index) {
+        const temp = { ...this.tasks[i], description };
+        this.tasks[i] = temp;
+      }
+    });
+  };
```
Comment (DeliceLydia) [inline src/Tasks.js:35]: - [ ] Edit function should edit only the selected task but see what is happening, here is a list of tasks
edit

see what happens when I edit one task it is actually changing all tasks , please address the issue
al

File: src/index.js [hunk 5]
```diff
@@ -1,24 +1,86 @@
...
+
   tasksList.innerHTML = newList;
+
+  alternateIcons();
+  editAction();
 };
 
+addInput.addEventListener('keypress', (e) => {
+  if (e.key === 'Enter') {
+    MyTasks.addTask(e.target.value, false);
```
Comment (DeliceLydia) [inline src/index.js:73]: - [ ] By default when you add a task property `completed` should be set to `false` but when you check a box it should be set to `true` but there is no change here please address that

Comment (DeliceLydia) [reply src/index.js:73]: - [ ] The button `clear all completed` should remove all completed tasks consider fixing that as well

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @nottherealalanturing, 

Good job so far! There are some issues that you still need to work on to go to the
next project but you are almost there! Highlights for your project:
- [ ] No linter errors
- [ ] Awesome README


## Required Changes ♻️
_Check the comments under the review._



### Keep Rocking!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: src/index.js (hunk 5 above)
Comment (nottherealalanturing) [reply src/index.js:73]: These functionalities are to be implemented in the next exercise.

**Comment by omar-labana** [comment]
# **Approved ✔️**

Hi @nottherealalanturing

Congratulations on completing this project! 👏🏼 🎉

You have correctly implemented all of the requirements📃 correctly and without any errors⚠️ or bugs🪲.

Cheers and good luck in your next milestones! 🍀

Feel free to leave any questions or comments in the PR thread(don't forget to tag me 😄) if something is not 100% clear.

---
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**